packages/fuels-test-helpers/src/utility.rs

```diff
@@ -0,0 +1,103 @@
+use std::future::Future;
+use std::time::Duration;
+
+pub async fn retry_until<Fut>(
+    condition: impl Fn() -> Fut,
```

Comment:
in our context, could we call this condition `predicate`? It would convey that it is actually a dynamic function that can change in the future.